source/greentea_test_io.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2013-2021, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "greentea-client/test_io.h"
+#include <stdio.h>
+
+/**
+ * \brief Read character from stream of data
+ *
+ *        Closure for default "get character" function.
+ *        This function is used to read characters from the stream
+ *        (default is serial port RX). Key-value protocol tokenizer
+ *        will build stream of tokes used by key-value protocol to
+ *        detect valid messages.
+ *
+ *        If EOF is received parser finishes parsing and stops. In
+ *        situation where we have serial port stream of data parsing
+ *        goes forever.
+ *
+ * \return Next character from the stream or EOF if stream has ended.
+ *
+ */
+int greentea_getc() {
+    return getchar();
+}
+
+
+/**
+ * \brief Write character from stream of data
+ *
+ * \return The number of bytes written
+ *
+ */
+void greentea_putc(int c) {
+    putchar(c);
+}
+
+/**
+ * \brief Write a string to the serial port
+ *
+ *        This function writes a '\0' terminated string from the target
+ *        to the host. It writes directly to the serial port using the
+ *        the write() method.
```

Comment:
Done. I've moved the docs into the header and kept them very minimal. Much of the old docs are about how the functions will be used instead of how they should behave.